sdk/eventhub/event-hubs/src/eventHubSender.ts

```diff
@@ -343,31 +343,55 @@ export class EventHubSender extends LinkEntity {
   }
   /**
    * Returns maximum message size on the AMQP sender link.
-   * @ignore
+   * @param abortSignal An implementation of the `AbortSignalLike` interface to signal the request to cancel the operation.
+   * For example, use the &commat;azure/abort-controller to create an `AbortSignal`.
    * @returns Promise<number>
+   * @throws {AbortError} Thrown if the operation is cancelled via the abortSignal.
    */
-  async getMaxMessageSize(): Promise<number> {
-    try {
-      if (!this.isOpen()) {
-        log.sender(
-          "Acquiring lock %s for initializing the session, sender and " +
-            "possibly the connection.",
-          this.senderLock
+  async getMaxMessageSize(abortSignal?: AbortSignalLike): Promise<number> {
+    return new Promise<number>(async (resolve, reject) => {
+      const rejectOnAbort = () => {
+        const desc: string = `[${this._context.connectionId}] The create batch operation has been cancelled by the user.`;
+        log.error(desc);
+        const error = new AbortError(`The create batch operation has been cancelled by the user.`);
+        reject(error);
+      };
+
+      const onAbort = () => {
+        abortSignal!.removeEventListener("abort", onAbort);
+        rejectOnAbort();
+      };
+
+      if (abortSignal) {
+        // the aborter may have been triggered between request attempts
+        // so check if it was triggered and reject if needed.
+        if (abortSignal.aborted) {
+          return rejectOnAbort();
+        }
+        abortSignal.addEventListener("abort", onAbort);
+      }
+      try {
+        if (!this.isOpen()) {
+          log.sender(
+            "Acquiring lock %s for initializing the session, sender and " +
+              "possibly the connection.",
+            this.senderLock
+          );
+          await defaultLock.acquire(this.senderLock, () => {
+            return this._init();
+          });
+        }
+        resolve(this._sender!.maxMessageSize);
+      } catch (err) {
+        log.error(
+          "[%s] An error occurred while creating the sender %s",
+          this._context.connectionId,
+          this.name,
+          err
         );
-        await defaultLock.acquire(this.senderLock, () => {
-          return this._init();
-        });
+        reject(err);
```

Comment:
After `init()` completes, regardless of whether it succeeds or fails, we should remove the abort event listener